File: Algorithms/Tree Traversal/Post-Order Traversal/postorder.cpp
// Prg for tree traversals in C++
#include <iostream>
using namespace std;

// A class for a node
struct Node
{
    int data;
    struct Node *left, *right;
    Node(int data)
    {
        this->data = data;
        left = right = NULL;
    }
};


// POSTORDER Traversal
void printPostorder(struct Node *node)
{
    if (node == NULL)
        return;
    printPostorder(node->left);  // recur on left child
    printPostorder(node->right);  // recur on right child
    cout << node->data << " ";  // print value
}


// MAIN
int main()
{
    struct Node *root = new Node(1);
    root->left = new Node(2);
    root->right = new Node(3);
    root->left->left = new Node(4);
    root->left->right = new Node(5);

    cout << "\nPostorder traversal of binary tree is \n";
    printPostorder(root);

    return 0;
}

File: Data Structures/Tree Traversal/Post-Order Traversal/postorder.js
// Prg for tree traversals in JS

function Node(data) {
    this.data = data;
    this.left = null;
    this.right = null;
}

// Creating Nodes
var root = new Node(1);
var n1 = new Node(2);
var n2 = new Node(3);
var n3 = new Node(4);
var n4 = new Node(5);

// Setting Children
root.left = n1;
root.right = n2;
n1.left = n3;
n1.right = n4;


// POSTORDER Traversal
function post_order(root, nodes) {
    if (root && root.left) {
        post_order(root.left, nodes);
    }
    if (root && root.right) {
        post_order(root.right, nodes);
    }
    console.log(root.data);
    return nodes;
}


// MAIN
console.log("PostOrder Traversal: ")
post_order(root, []);